server/services/topics.ts

```diff
@@ -37,7 +37,7 @@ const indexMethod: Method = {
 async function index({ query }: { query: IndexQuery }) {
   const page = query.page ?? 0;
   const perPage = query.per_page ?? 100;
```

Comment:
100件ごとの取得、画面によってはそれぞれの要素で動画プレイヤーを表示するので、私のローカル環境だと10秒くらい待つ必要があり重いですね…！
このPRで検討すべきことではないかもしれないですが、マイトピックだと3列*6行で18件ずつとかだと無限スクロールが快適にできるのかな？など思いました
```suggestion
  const perPage = query.per_page ?? 18;
```

Reply:
なるほど。
たしかに大量に動画プレイヤーがロードされるのでメモリー量的に環境に依っては厳しい感じがしますね。
ちなみにそのような環境でたとえば100-1000件くらいのトピックが並ぶとどんな感じになるんでしょうか。

Reply:
私の環境でも300件くらいでメモリ1GB超えますねw

Reply:
dev server だからだろうか。export して実験してみます。

Reply:
本番相当でexportしてみると100件のとき使用量 400MB くらいまで減りましたがなかなか厳しい…

Reply:
メモリー量が厳しそうなのもそうですが、動画プレイヤーがロードされて再生準備完了になるまでの再描画の負荷も体感に影響しているかもしれません

Reply:
> たとえば100-1000件くらいのトピックが並ぶとどんな感じになるんでしょうか。

devtoolsでなにか共有できそうだと思うのでお昼食べたらみてみます！

Reply:
Lighthouse (デスクトップ)

![Screenshot from 2021-02-18 12-46-47](https://user-images.githubusercontent.com/1730234/108302565-910f1180-71e7-11eb-8a29-6154ababbec0.png)
![Screenshot from 2021-02-18 12-47-10](https://user-images.githubusercontent.com/1730234/108302570-92403e80-71e7-11eb-85c7-a0aaab279164.png)

Reply:
intersection observerでrootMarginを広めにして、領域外では動画プレイヤーを破棄するような処理を追加するとよいのでしょうか

Reply:
100件は多すぎる。よって減らす。
デフォルトを18件にしてみるとたしかにパフォーマンス改善: 5→13、FCP6.2→4s、TtoI25.8→5.6s
しかし依然としてレッドなので根本解決にはならないように感じた。
とりあえずサーバー側では半分50件をデフォルトに修正、またトピック一覧、マイトピック一覧についてはいずれもクライアント側のAPIの呼び出しの部分で18件に指定することにします。

Reply:
もしグリーンを目指すならば、一切の動画プレイヤーを読み込まない、という選択肢が現実的な気がする。
あるいは無限スクロールではなく単純なページネーションにすべきだったかもしれない。反省。

Reply:
> intersection observerでrootMarginを広めにして、領域外では動画プレイヤーを破棄するような処理を追加するとよいのでしょうか

たしかにそれもよいかもしれませんね! なるほど。